core/src/main/java/io/grpc/Deadline.java

```diff
@@ -141,4 +141,14 @@ public long timeRemaining(TimeUnit unit) {
   public String toString() {
     return timeRemaining(TimeUnit.NANOSECONDS) + " ns from now";
   }
+
+  @Override
+  public int compareTo(Deadline that) {
+    if (this.deadlineNanos < that.deadlineNanos) {
```

Comment:
Hmm.  You are right that one should subtract the other, but from the docs it should be the later one.  Which one is the later one?  Normal interaction looks like:

    t1 = System.nanoTime();
    ...
    t2 = System.nanoTime();
    diff = t2 - t1;

One possibility would be to do something like:

    now =  System.nanoTime();
    remainingThis = now - this.deadlineNanos;
    remainingThat = now - that.deadlineNanos;
    if (remainingThis > remainingThat) {
      ...
    }  

thoughts?